src/api/resources/fish_species.py

```diff
@@ -67,9 +74,16 @@ class FishSpeciesViewSet(BaseAttributeApiViewSet):
     serializer_class = FishSpeciesSerializer
     queryset = (
         FishSpecies.objects.select_related()
-        .extra(select={"display_name": "fish_genus.name || ' ' || fish_species.name"})
-        .prefetch_related(Prefetch("regions", queryset=Region.objects.all().only("id")))
+        .annotate(
+            regions_=ArrayAggExt(
+                "regions",
+                # filter=Q(regions__isnull=False),
+                # default=Value([])
+            ),
+            display_name=Concat(F("genus__name"), Value(" "), F("name"))
```

Comment:
We still have to deal with `FishGenus` and `FishFamily` right?

Reply:
didn't bother, it seemed reasonably quick